### ISSUE TITLE
Add context state enhancers

### ISSUE THREAD

**Issue by diegohaz** [diegohaz/reas#39]
I know very few about the new React Context API. But I wonder if we can make something like `withStepContext`, with the same interface as `withStepState`, but providing global state. It would be an alternative to `withStepReducer` (planned on #8):

```jsx
import React from 'react'
import { Step, withStepState, withStepContext, withStepReducer } from 'reas'

const MySteps = ({ step }) => (
  <div>
    <Step step="foo" {...step}>Foo</Step>
    <Step step="bar" {...step}>Foo</Step>
    <Step step="baz" {...step}>Foo</Step>
  </div>
)

// today
export default withStepState(MySteps)
// or maybe
export default withStepContext(MySteps)
// or maybe
export default withStepReducer(MySteps)
```

UPDATE:

Since we've replaced state enhancers HOCs with render props in #47, this is the new way I'm planning to have global state:
```jsx
import React from 'react'
import { Step } from 'reas'

const MySteps = () => (
  <Step.State global="foo">
    {step => (
      <div>
        <Step step="foo" {...step}>Foo</Step>
        <Step step="bar" {...step}>Foo</Step>
        <Step step="baz" {...step}>Foo</Step>
      </div>
    )}
  </Step.State>
)
```

**Comment by diegohaz** [comment]
Created a list of resources related to the new context API: https://github.com/diegohaz/awesome-react-context

**Comment by diegohaz** [comment]
Closed in #48
